geode-core/src/main/java/org/apache/geode/distributed/internal/FunctionExecutionPooledExecutor.java

```diff
@@ -91,38 +91,17 @@ private static RejectedExecutionHandler initREH(final BlockingQueue<Runnable> q,
       boolean forFnExec) {
     if (forFnExec) {
       return new RejectedExecutionHandler() {
-        public void rejectedExecution(final Runnable r, ThreadPoolExecutor executor) {
+        public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
           if (executor.isShutdown()) {
             throw new RejectedExecutionException(
                 "executor has been shutdown");
           } else {
-            // The request was rejected due to all threads being in use.
-            //
-            // In the normal case, add the rejected request to the blocking queue.
-            //
-            // If the handler was invoked via the bufferConsumer (the thread that takes from the
-            // blocking queue and offers to the synchronous queue), then spin off a thread directly.
-            // In this case, an offer has already been made to the synchronous queue and failed, so
-            // all the function execution threads are in use.
-            //
-            // If the handler was invoked via a function execution thread, then spin off a thread
-            // directly. In this case, that means a function is executing another function. The
-            // child function request shouldn't be in the queue behind the parent request since the
-            // parent function is dependent on the child function executing.
-            final boolean isBufferConsumer = isBufferConsumer(executor);
-            if (isBufferConsumer || isFunctionExecutionThread()) {
-              if (isBufferConsumer) {
-                logger.warn("An additional " + FUNCTION_EXECUTION_PROCESSOR_THREAD_PREFIX
-                    + " thread is being launched because all " + executor.getMaximumPoolSize()
-                    + " thread pool threads are in use for greater than " + OFFER_TIME + " ms");
-              } else {
-                if (logger.isDebugEnabled()) {
-                  logger.warn("An additional " + FUNCTION_EXECUTION_PROCESSOR_THREAD_PREFIX
-                      + " thread is being launched to prevent slow performance due to nested function executions");
-                }
-              }
-              launchAdditionalThread(r, executor);
+            if (isBufferConsumer(executor)) {
```

Comment:
feel free not to make this change, but...
you could combine line 98 and 99 into:
  } else if {
and get rid of an extra layer of indentation.